src/Microsoft.AspNetCore.SignalR.Client.Core/InvocationRequest.cs

```diff
@@ -133,10 +145,24 @@ public NonStreaming(CancellationToken cancellationToken, Type resultType, string
 
             public Task<object> Result => _completionSource.Task;
 
-            public override void Complete(object result)
+            public override void Complete(CompletionMessage completionMessage)
             {
+                if (completionMessage.IsStreamingCompletion)
+                {
+                    Logger.ReceivedUnexpectedStreamingCompletion(InvocationId);
+                    _completionSource.TrySetException(new InvalidOperationException("Server returned streaming completion message for a non-streamed invocation."));
+                    return;
+                }
+
+                if (!string.IsNullOrEmpty(completionMessage.Error))
+                {
+                    Fail(new HubException(completionMessage.Error));
```

Comment:
Should the error message be some non-empty, descriptive string instead?

Reply:
It is. This is the error thrown on the server side.